### ISSUE TITLE
Disable SSL verification

### ISSUE THREAD

**Issue by TheMatrix97** [cdent/gabbi#273]
Hi! I'm using gabbit to build/execute my integration tests towards a django API, which has a self signed ssl certificate. 
I'm getting the following error from gabbi
`urllib3.exceptions.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:777)` 
I'm pretty sure that's caused because my gunicorn certificate is self signed / signed by a non trusted CA
Also I saw that this could be solved if there was a way to create a custom HTTPS connection poll
`import urllib3
c = urllib3.HTTPSConnectionPool(IP, port=PORT, cert_reqs='CERT_NONE', assert_hostname=False)`
How could I run my tests without SSL verification? 
Thank you in advance!

**Comment by cdent** [comment]
If you can try something, and it verifies as working, we can try to come up with a generic solution for passing when creating the tests. But the thing to try in the meantime is adjusting the gabbi code so the arguments passed in when creating a pool manager include the ConnectionPool args you need. That would be done in [get_http](https://github.com/cdent/gabbi/blob/master/gabbi/httpclient.py#L173) and the args would be added to the calls to instantiate an `Http` and `VerboseHttp`.

So I think what would be added is one or both of `cert_reqs='CERT_NONE', assert_hostname=False`.

If you can help determine which or both are needed, we can figure out the next step from there.

**Comment by TheMatrix97** [comment]
Finally I did some monkey patching to solve my problem
```python
from gabbi import httpclient
import certifi
import sys
from gabbi.httpclient import Http
from gabbi.httpclient import VerboseHttp

def run():
    print("changing HTTP client...")
    httpclient.get_http = get_http
    return

def get_http(verbose=False, caption=''):
    """Return an ``Http`` class for making requests."""
    if not verbose:
        return Http(strict=True, ca_certs=certifi.where(), cert_reqs='CERT_NONE')

    headers = False if verbose == 'body' else True
    body = False if verbose == 'headers' else True
    return VerboseHttp(headers=headers, body=body, stream=sys.stdout,
                       caption=caption, colorize=True, strict=True,
                       ca_certs=certifi.where(), cert_reqs='CERT_NONE')
```
I execute the run function before calling the `driver.build_tests` function 
Seems that with just disable the ssl verification (cert_reqs=None) is enough.

Thank you very much for your help, I'm still thinking that an ssl verification disable param would be a nice feature to add, as like me, there are lots of people who use self signed certificates in their development servers.
I'll take a look to the source code and try to come up with a generic solution :)

**Comment by cdent** [comment]
Great, thanks for verifying that. Given that, I think the right fix would be to add a test attribute, similar to `verbose`, something like `cert_validate` that defaults to `True`. When `False`, `cert_reqs='CERTS_NONE'` would be set when creating the http client. This could be managed per test, set in the `defaults` section of the gabbi YAML files, or set when building tests or using gabbi-run.

If you're happy to try to make this go that would be great. If you agree that the way verbose works is a good model to follow, the relevant parts of the code that may be a useful guide include:

* https://github.com/cdent/gabbi/blob/c8244db323af4591a80e19915d461451d4ad3170/gabbi/driver.py#L129-L133
* https://github.com/cdent/gabbi/blob/c8244db323af4591a80e19915d461451d4ad3170/gabbi/case.py#L60
* https://github.com/cdent/gabbi/blob/c8244db323af4591a80e19915d461451d4ad3170/gabbi/httpclient.py#L173
* https://github.com/cdent/gabbi/blob/c8244db323af4591a80e19915d461451d4ad3170/gabbi/runner.py#L238-L243
* https://github.com/cdent/gabbi/blob/c8244db323af4591a80e19915d461451d4ad3170/gabbi/suitemaker.py#L82-L83

Thanks for reporting this, and for being interested in making gabbi better. For sake of history: The reason this wasn't already possible is that most of the environment in which gabbi is used either the testing is over `wsgi-intercept` so SSL doesn't come into it, or with integration tests the services aren't running with SSL. I agree that there are plenty of people who want to do that, and gabbi needs to support that.

**Comment by TheMatrix97** [comment]
I agree that the most efficient way to do testing is using wsgi-interceptor, but I found an error when using it with django 2.2 ... I think the best will be create another issue and keep this one clean
Thanks for the guide, I'll take a look and hopefully send a merge request with the solution ;)

**Comment by cdent** [comment]
> but I found an error when using it with django 2.2 ... I think the best will be create another issue

Yeah, do that please. I would guess that the problem is in http://github.com/cdent/wsgi-intercept itself, but can't really be sure until there's more info.

**Comment by TheMatrix97** [comment]
Hey @cdent,
Sorry for the delay, here is my pull request -> #274 
I added the cert_validate attribute at `py_test_generator` and `build_tests` functions. Furthermore I added a param `--no-cert-validate` to the gabbi-run which adds the attribute `req_certs='CERT_NONE` to the http(s) client.

**Comment by cdent** [comment]
Fixed by #274 and released in 1.49.0
